src/main/java/genie/Ui.java

```diff
@@ -0,0 +1,166 @@
+package genie;
+
+import genie.task.Task;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Deals with interactions with the user.
+ */
+public class Ui {
+
+    // Art generated from: https://patorjk.com/
+    private final static String GENIE_LOGO =
+                    "                       ______                  _       \n" +
+                    "                      / ____/  ___    ____    (_)  ___ \n" +
+                    "                     / / __   / _ \\  / __ \\  / /  / _ \\\n" +
+                    "                    / /_/ /  /  __/ / / / / / /  /  __/\n" +
+                    "                    \\____/   \\___/ /_/ /_/ /_/   \\___/ \n";
+    /**
+     * A constructor for Ui class.
+     */
+    public Ui() {}
+
+    /**
+     * Prints greet message.
+     */
+    public void greet() {
+        System.out.println("Hello! This is Genie, your personal task tracker!");
+    }
+
+    /**
+     * Prints Genie logo.
+     */
+    public void bootLogo() {
+        printLine();
+        System.out.println(GENIE_LOGO);
+        printLine();
+    }
+
+    /**
+     * Prints a line.
+     */
+    public void printLine() {
+        System.out.println("==============================================================================");
+    }
+
+    /**
+     * Prints task list if it is not empty. Otherwise, prints an empty task message.
+     * @param tasks
+     */
+    public void printList(ArrayList<Task> tasks) {
+        if (tasks.size() > 0) {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 1; i <= tasks.size(); i++) {
+                Task t = tasks.get(i - 1);
+                System.out.println(i + ". " + t.toString());
+            }
+        } else {
+            showEmptyListMessage();
+        }
+        printLine();
+    }
+
+    /**
+     * Echos user's input.
+     * @return user's input
+     */
+    public String readCommand() {
+        Scanner sc = new Scanner(System.in);
+        return sc.nextLine();
+    }
+
+    /**
+     * Prints mark done message.
+     * @param t task to be marked done
+     */
+    public void showMarkDoneMessage(Task t) {
+        System.out.println("Nice! I've marked this task as done:\n" + "  " + t.toString());
+        printLine();
+    }
+
+    /**
+     * Prints add task message.
+     * @param t task to be added
+     * @param size number of tasks in the list
+     */
+    public void showAddTaskMessage(Task t, int size) {
+        System.out.println("Got it. I've added this task:\n  " + t.toString());
```

Comment:
Try to follow the "SLAP Hard" principle, and make the other 2 printing as methods as well?